File: itlen/app/src/main/java/com/deadely/itl_en/di/module/ActivityModule.java
package com.deadely.itl_en.di.module;

import androidx.appcompat.app.AppCompatActivity;

import dagger.Module;

@Module
public class ActivityModule {
    private AppCompatActivity appCompatActivity;

    public ActivityModule(AppCompatActivity appCompatActivity) {
        this.appCompatActivity = appCompatActivity;
    }

    public AppCompatActivity getAppCompatActivity() {
        return appCompatActivity;
    }
}

File: itlen/app/src/main/java/com/deadely/itl_en/di/module/ActivityModule.kt
package com.deadely.itl_en.di.module

import androidx.appcompat.app.AppCompatActivity
import dagger.Module
import dagger.Provides

@Module
class ActivityModule(private val appCompatActivity: AppCompatActivity) {
    @Provides
    fun provideActivity(): AppCompatActivity {
        return appCompatActivity
    }
}